File: coupon/coupon-api/src/main/java/dev/be/coupon/api/coupon/application/dto/CouponCreateCommand.java
package dev.be.coupon.api.coupon.application.dto;


import dev.be.coupon.domain.coupon.CouponDiscountType;
import dev.be.coupon.domain.coupon.CouponType;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

public record CouponCreateCommand(
        UUID userId,
        String couponName,
        CouponType couponType,
        CouponDiscountType couponDiscountType,
        BigDecimal couponDiscountValue,
        int totalQuantity,
        LocalDateTime expiredAt
) {
}

File: coupon/coupon-api/src/main/kotlin/dev/be/coupon/api/coupon/application/dto/CouponCreateCommand.kt
package dev.be.coupon.api.coupon.application.dto

import dev.be.coupon.domain.coupon.Coupon
import dev.be.coupon.domain.coupon.CouponTypeConverter
import java.time.LocalDateTime
import java.util.*

data class CouponCreateCommand(
    val userId: UUID,
    val name: String,
    val type: String,
    val totalQuantity: Int,
    val validFrom: LocalDateTime,
    val validUntil: LocalDateTime
) {
    fun toDomain(): Coupon {
        return Coupon(
            name,
            CouponTypeConverter.from(type),
            totalQuantity,
            validFrom,
            validUntil
        )
    }
}
